providers/Mora/mod.ts

```diff
@@ -0,0 +1,251 @@
+import { ApiArgs, MediaFormat, PackageMeta, Track, WithApiUrl } from './json_types.ts';
+import type {
+	ArtistCreditName,
+	Artwork,
+	EntityId,
+	HarmonyEntityType,
+	HarmonyRelease,
+	HarmonyTrack,
+	LinkType,
+	ReleaseGroupType,
+} from '@/harmonizer/types.ts';
+import { type CacheEntry, MetadataProvider, ReleaseLookup } from '@/providers/base.ts';
+import { DurationPrecision, FeatureQuality, FeatureQualityMap } from '@/providers/features.ts';
+import { parseISODateTime, PartialDate } from '@/utils/date.ts';
+import { ProviderError, ResponseError } from '@/utils/errors.ts';
+import { extractMetadataTag } from '@/utils/html.ts';
+import { isValidGTIN } from '../../utils/gtin.ts';
+
+export default class MoraProvider extends MetadataProvider {
+	readonly name = 'Mora';
+
+	readonly supportedUrls = new URLPattern({
+		hostname: 'mora.jp',
+		pathname: '/package/:labelCode([0-9]+)/:materialNo{/}?',
+	});
+
+	readonly artistUrlPattern = new URLPattern({
+		hostname: this.supportedUrls.hostname,
+		pathname: '/artist/:id{/}?',
+	});
+
+	override readonly features: FeatureQualityMap = {
+		// The API returns a "full size" image of at most 200x200
+		'cover size': 200,
+		'duration precision': DurationPrecision.SECONDS,
+		'GTIN lookup': FeatureQuality.MISSING,
+		'MBID resolving': FeatureQuality.PRESENT,
```

Comment:
Not present yet, but this will work once we construct artist IDs which can be resolved.